### PR TITLE
Add support for other data types

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/Npgsql.PlDotNET/NpgsqlDataReader.cs [hunk 1]
```diff
@@ -112,12 +91,9 @@ public override bool Read()
                 this.ColumnNames = columnNamePts.ToList().Select(namePts => Marshal.PtrToStringAuto(namePts)).ToArray();
             }
```
Comment (ghost) [inline src/Npgsql.PlDotNET/NpgsqlDataReader.cs:92]: We should move this part to the constructor... do you agree, @jose-mario-jr ?

File: src/Npgsql.PlDotNET/NpgsqlDataReader.cs [hunk 2]
```diff
@@ -143,61 +129,82 @@ public override int GetOrdinal(string name)
...
+        // public string getTableLinesMd()
+        // {
+        //     var sb = new System.Text.StringBuilder();
 
-            return sb.ToString();
-        }
+        //     foreach (DataRow dataRow in this.ReturnedTable.Rows)
+        //     {
+        //         sb.Append($"| {string.Join(" | ", dataRow.ItemArray)} |\n");
+        //     }
```
Comment (ghost) [inline src/Npgsql.PlDotNET/NpgsqlDataReader.cs:173]: @jose-mario-jr, as we are using concepts of lazy evaluation now, we don't have this table with the .net values. How should we implement the ToString method? How does Npgsql do this?

File: src/Npgsql.PlDotNET/NpgsqlDataReader.cs (hunk 1 above)
Comment (jose-mario-jr) [reply src/Npgsql.PlDotNET/NpgsqlDataReader.cs:92]: I agree, it makes more sense
but
We only read the first line in this `Read` function, in the constructor we do not have the columnNamePts yet, so we need to refactor to do some kind of pre-fetch of the table or something like that before the `Read` function

File: src/Npgsql.PlDotNET/NpgsqlDataReader.cs (hunk 2 above)
Comment (jose-mario-jr) [reply src/Npgsql.PlDotNET/NpgsqlDataReader.cs:173]: It doesn't implement
this method is just to override the default behaviour of the `ToString` function that returns the object type, I implemented this function as a way to debug only, we can even get rid of it when we have the correct way to retrieve the values

**Pull request merged**